src/Microsoft.Identity.Client/Features/TokenCache/TokenCacheAccessor.cs

```diff
@@ -88,9 +98,22 @@ public ICollection<AccessTokenCacheItem> GetAllAccessTokens(string clientId)
             return returnList.Where(t => t.ClientId.Equals(clientId)).ToList();
         }
 
+        public ICollection<string> GetAllAccessTokensAsString()
+        {
+            return
+                new ReadOnlyCollection<string>(
+                    TokenCacheDictionary.Values.Where(
+                        v =>
+                            (JsonHelper.DeserializeFromJson<AccessTokenCacheItem>(v).Scope != null) &&
```

Comment:
you are already doing the deserialize here, can you return the collection of access tokens?